.github/workflows/build.yml

```diff
@@ -23,24 +23,43 @@ jobs:
         with:
           node-version: ${{ matrix.node }}
           cache: yarn
-      - run: yarn
+      - run: yarn install --ignore-scripts
+      - run: yarn build:cjs
       - run: tar -czf /tmp/web3-${{ matrix.node }}.js.tar.gz --exclude="./.git" ./
       - uses: actions/upload-artifact@v3
         with:
           name: web3-${{ matrix.node }}.js.tar.gz
           path: /tmp/web3-${{ matrix.node }}.js.tar.gz
-
+  build-cjs-types:
+      name: Build ESM & Types
+      needs: build
+      runs-on: ubuntu-latest
+      strategy:
+          matrix:
+              node: [ 18, 16 ]
+      steps:
+          -   uses: actions/download-artifact@v3
+              with:
+                  name: web3-${{ matrix.node }}.js.tar.gz
+                  path: /tmp
+          -   run: tar -xf /tmp/web3-${{ matrix.node }}.js.tar.gz -C ./
+          -   run: yarn build:esm
+          -   run: yarn build:types
   lint:
-    name: lint
-    needs: build
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/download-artifact@v3
-        with:
-          name: web3-14.js.tar.gz
-          path: /tmp
-      - run: tar -xf /tmp/web3-14.js.tar.gz -C ./
-      - run: yarn lint
+      name: Lint
+      needs: build
+      runs-on: ubuntu-latest
+      strategy:
+          matrix:
+              node: [ 18 ]
+      steps:
+          -   uses: actions/download-artifact@v3
+              with:
+                  name: web3-${{ matrix.node }}.js.tar.gz
+                  path: /tmp
+          -   run: tar -xf /tmp/web3-${{ matrix.node }}.js.tar.gz -C ./
+          -   run: test -f ./.secrets.json || echo '{"SEPOLIA":{"HTTP":"","WS":"","ACCOUNT":{"address":"0x0000000000000000000000000000000000000000","privateKey":""},"ALLOWED_SEND_TRANSACTION":false},"MAINNET":{"HTTP":"","WS":"","ACCOUNT":{"address":"0x0000000000000000000000000000000000000000","privateKey":""},"ALLOWED_SEND_TRANSACTION":false}}' > ./.secrets.json
```

Comment:
The `.secrets` file is created after building the project using the [prebuild NPM script](https://github.com/web3/web3.js/blob/e550a9c88edec819a8ac0181fc2d321f8e959121/package.json#L36) which calls the [init.ts script](https://github.com/web3/web3.js/blob/4.x/scripts/init.ts)

Reply:
Thanks! I will change it properly

Reply:
fixed